rfcs/20190225-tf-on-demand.md

```diff
@@ -0,0 +1,191 @@
+# TensorFlow On Demand
+
+| Status        | Proposed      |
+:-------------- |:---------------------------------------------------- |
+| **Author(s)** | Gunhan Gulsoy (gunan@google.com), Hye Soo Yang (hyey@google.com) |
+| **Sponsor**   | Gunhan Gulsoy (gunan@google.com)                |
+| **Collaborators** | Christoph Goern (goern@redhat.com), Subin Modeel (smodeel@redhat.com)         |
+| **Updated**   | 2019-02-25                                           |
+
+## Objective
+
+This document proposes a system to build optimized binaries in the cloud and deliver the artifacts to users of TensorFlow. With this system, we aim to achieve the following:
+
+*   Hide the complexity of the build system of TF from beginner and intermediate users.
+*   Improve the out-of-the-box performance for users.
+*   Build on the strength of TensorFlow's community partners.
+
+## Motivation
+
+### Overview
+
+TF build is difficult and the friction present during building phase is apparent in our user experiences. For a successful build, users needs to be aware of requirements and configurations that they might not be too familiar with. This can be extremely challenging for beginner and intermediate users.
+
+Currently, TF python release artifacts are hosted on PyPI as a single source of TF binaries for download. However, PyPI is quite limited in capability; it is unable to recognize and/or deploy on user machines.
```

Comment:
Does this proposal mean that TensorFlow binaries will no longer be released on PyPI?

Reply:
Guess not.

> Does this proposal mean that TensorFlow binaries will no longer be released on PyPI?

Guess not.

Reply:
These builds are for specific combination of OS/python/CUDA/arch etc. PyPI currently lacks the capability to differentiate wheels based on platform.
PyPi will continue to have the tensorflow wheels with generic arch.

Reply:
From my point of view its worth to think about publishing TF Wheels in a different thread: there are a few issues like naming (mentioned by Subin above), include build time configuration in wheel file, ...

I am sure we can not change the way PyPi works short term, so we need to come up with a cleaver structure of index hosts and directories.